app-modules/service-management/src/Notifications/ServiceRequestAssigned.php

```diff
@@ -0,0 +1,97 @@
+
+    Aiding App™ is licensed under the Elastic License 2.0. For more details,
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\Notifications;
+
+use AidingApp\Notification\Notifications\BaseNotification;
+use AidingApp\Notification\Notifications\Channels\DatabaseChannel;
+use AidingApp\Notification\Notifications\Channels\EmailChannel;
+use AidingApp\Notification\Notifications\Concerns\DatabaseChannelTrait;
+use AidingApp\Notification\Notifications\Concerns\EmailChannelTrait;
+use AidingApp\Notification\Notifications\DatabaseNotification;
+use AidingApp\Notification\Notifications\EmailNotification;
+use AidingApp\Notification\Notifications\Messages\MailMessage;
+use AidingApp\ServiceManagement\Filament\Resources\ServiceRequestResource;
+use AidingApp\ServiceManagement\Models\ServiceRequest;
+use App\Models\NotificationSetting;
+use Filament\Notifications\Notification;
+
+class ServiceRequestAssigned extends BaseNotification implements EmailNotification, DatabaseNotification
+{
+    use EmailChannelTrait;
+    use DatabaseChannelTrait;
+
+    /**
+     * @param class-string $channel
+     */
+    public function __construct(
+        public ServiceRequest $serviceRequest,
+        public string $channel,
+    ) {}
+
+    /**
+     * @return array<int, string>
+     */
+    public function via(object $notifiable): array
+    {
+        return [match ($this->channel) {
+            DatabaseChannel::class => 'database',
+            EmailChannel::class => 'mail',
+        }];
```

Comment:
Hmm, I wonder if perhaps we want to consider `channel` becoming `channels` which accepts an array of channels?

That way you only have to dispatch a single notification that creates jobs for the correct channels for you.

Reply:
A user could be a manager, or an auditor, or both, you would then have to do logic to determine which groups they are part of and dispatch a different notification object based on that. If you check out the code that queries the managers or auditors I think this is the most efficient way to avoid that, so you don't have to combine the query results. What do you think?

Reply:
Hmm, I see your point but let's chat about it all a little more. Perhaps at office hours.